core/camel-core-languages/src/main/java/org/apache/camel/language/csimple/CSimpleHelper.java

```diff
@@ -163,6 +178,40 @@ public static String bodyOneLine(Exchange exchange) {
         return body;
     }
 
+    public static String prettyBody(Exchange exchange) {
+        String body = exchange.getIn().getBody(String.class);
+
+        if (body == null) {
+            return null;
+        } else if (body.startsWith("{") && body.endsWith("}") || body.startsWith("[") && body.endsWith("]")) {
+            body = Jsoner.prettyPrint(body); //json
+        } else if (body.startsWith("<") && body.endsWith(">")) {
+            return CSimpleHelper.prettyXml(body); //xml
+        }
```

Comment:
You should rather rely on the content type (aka the header `Exchange.CONTENT_TYPE`)

Reply:
is this header always set? What if user load data from a file?